### PR TITLE
QL: Join-order query fixes

### PR DESCRIPTION
**Pull request by MathiasVP**
This PR removes the "bucket" terminology that I stole from DCA. In this case, a "bucket" is really a pipeline ordering, so there's no reason to use that terminology here.

### CODE REVIEW

File: ql/ql/src/queries/performance/LargeJoinOrder.ql [hunk 1]
```diff
@@ -130,9 +129,9 @@ int getSize(ComputeRecursive recursive, string predicateName, int iteration, TDe
     kind = TCurrent() and
     i = iteration
   |
-    result = getInLayerEventWithName(recursive, predicateName).getDeltaSize(iteration - 1)
+    result = getInLayerEventWithName(recursive, predicateName).getDeltaSize(i)
     or
-    not exists(getInLayerEventWithName(recursive, predicateName).getDeltaSize(iteration - 1)) and
+    not exists(getInLayerEventWithName(recursive, predicateName).getDeltaSize(i)) and
```
Comment (MathiasVP) [inline ql/ql/src/queries/performance/LargeJoinOrder.ql:134]: This is a small drive-by fix. We were incorrectly using `iteration - 1` here (which was only correct when `kind = TPrevious()`).

**Review by tausbn — APPROVED** [review]
